File: 539-minimum-time-difference/minimum-time-difference.py
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        def getMinutes(s):
            time = s.split(":")
            return int(time[0]) * 60 + int(time[1])

        times, points = set(), []
        for i in timePoints:
            s = getMinutes(i)
            if s in times:
                return 0
            times.add(s)
            points.append(s)

        points.sort()
        m = 24 * 60 - points[-1] + points[0]
        for i in range(len(points) - 1):
            m = min(m, points[i + 1] - points[i])
        return m

File: 539-minimum-time-difference/minimum-time-difference.java
class Solution {
    public int getMinutes(String time) {
        int hour = Integer.parseInt(time.substring(0, 2));
        int mins = Integer.parseInt(time.substring(3, 5));
        return hour * 60 + mins;
    }
    public int findMinDifference(List<String> timePoints) {
        List<Integer> times = new ArrayList<Integer>();
        for (int i = 0; i < timePoints.size(); i++) {
            Integer s = getMinutes(timePoints.get(i));
            if (times.contains(s))
                return 0;
            times.add(s);
        }
        
        Collections.sort(times);
        int m = 1440 - times.get(times.size() - 1) + times.get(0);
        for (int i = 0; i < times.size() - 1; i++) 
            m = Math.min(m, times.get(i + 1) - times.get(i));

        return m;
    }
}
